In [1]:
# Setup imports and short-form names for each of the packages needed for data processing and visualizations
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns 
%matplotlib inline

In [12]:
data = pd.read_csv('results.csv', dtype={'id': str, 'startTime': 'Int64', 'origin': object, 'destination': object, 'totalPower': float, 'minimumCapacity': int, 'chargeLevelInterval': int, 'optimizedCost': float, 'optimizedCostDuration': float, 'optimizedDuration': float, 'optimizedDurationFinancialCost': float, 'endTime': 'Int64', 'error': object})
data.index.name = 'Index'
data.head()

id      startTime  \
Index                               
0      f3eed27de2f  1684487402321   
1      e50030aca2d  1684487461302   
2      8b04265f66d  1684487650236   
3      c961b63aa5f  1684489081631   
4      fbdc3f71999  1684490882000   

                                                  origin  \
Index                                                      
0      {"latitude":59.188956929696936,"longitude":13....   
1      {"latitude":60.23301952977527,"longitude":13.5...   
2      {"latitude":59.74999614778409,"longitude":15.2...   
3      {"latitude":55.887033121156385,"longitude":13....   
4      {"latitude":59.74999614778409,"longitude":15.2...   

                                             destination    totalPower  \
Index                                                                    
0      {"latitude":59.94046649336293,"longitude":13.6...   9245.573059   
1      {"latitude":59.60549803405616,"longitude":16.7...  28294.802206   
2      {"latitude":56.75573491879138,"longitude":15.4...  41585.574499   
3      {"latitude":60.14050023753482,"longitude":13.1...  68059.786133   
4      {"latitude":56.75573491879138,"longitude":15.4...  41585.574499   

       minimumCapacity  chargeLevelInterval  optimizedCost  \
Index                                                        
0                   43                   50            0.0   
1                   43                   50            NaN   
2                   43                   50            NaN   
3                   43                   50            NaN   
4                   43                   50            NaN   

       optimizedCostDuration  optimizedDuration  \
Index                                             
0                     5700.0             5700.0   
1                        NaN                NaN   
2                        NaN                NaN   
3                        NaN                NaN   
4                        NaN                NaN   

       optimizedDurationFinancialCost        endTime error  
Index                                                       
0                                 0.0  1684487461279   NaN  
1                                 NaN  1684487650186   NaN  
2                                 NaN           <NA>   NaN  
3                                 NaN           <NA>   NaN  
4                                 NaN           <NA>   NaN

In [13]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 349 entries, 0 to 348
Data columns (total 13 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              349 non-null    object 
 1   startTime                       349 non-null    Int64  
 2   origin                          349 non-null    object 
 3   destination                     349 non-null    object 
 4   totalPower                      349 non-null    float64
 5   minimumCapacity                 349 non-null    int64  
 6   chargeLevelInterval             349 non-null    int64  
 7   optimizedCost                   197 non-null    float64
 8   optimizedCostDuration           197 non-null    float64
 9   optimizedDuration               165 non-null    float64
 10  optimizedDurationFinancialCost  165 non-null    float64
 11  endTime                         175 non-null    Int64  
 12  error                           0 no

## Out of 349 routes, 165 were completed without running out of memory